### ISSUE TITLE
Question regarding normalization

### ISSUE THREAD

**Issue by iamgroot42** [Cabbagito/Fine-Tuning-Whisper-on-LibriSpeech#1]
First off, thanks a lot for the implementation! Looking at the Librispeech dataset, I realize that the ground truth is in all caps, whereas model predictions (before fine-tuning) are mostly lower case. While this is not an issue for WER computation (since it involves normalization), lower and upper case texts get tokenized very differently, and I wonder if this hinders the model training process _i.e._ should the ground-truth text also be sent to lower-case before it is fed to the model for loss computation (and subsequent backprop).

I ask because in my own attempt to fine-tune on LibriSpeech, I notice that the WER jumps from ~5 at the beginning to 15-16 and then drops to 9 as training progresses, whereas the loss function converges smoothly. I wonder if this is an artifact arising from the model suddenty trying to learn to predict everything in all-caps?

**Comment by Cabbagito** [comment]
Hi, 
I agree, it makes more sense to feed the model uncapitalized text, since then it only has to learn to not capitalize words instead of having to capitalize everything. But also once the decoder gets the first token in capitalized form, it should continue that trend.

Regarding your attempt, the jump in WER seems quite substantial. I also sent the capitalized text to the model, but the WER hovered around 5%. You can see this from the training plot in my project. 
It is quite possible that the culprit for this jump is the fact that the model has to learn to generate everything in capitalized form. One experiment I wanted to try, that I didn't do at the time of implementing this project, was to only train the decoder and head, or the last couple of layers of the decoder. If you are training the whole model, this might make the output unstable because you are also changing the encoder. 
Also, keep in mind that the LibriSpeech data also doesn't contain punctuation, and you should consider adding it as a normalization step if you haven't already.

But still, this jump in WER does seem fishy. I would double check the code, potentially there is a problem with the normalization implementation since the loss is converging but the WER score gets worse.

Hope this helps, and if you have any other questions feel free to ask.

P.S.
One more thing I realized while implementing the project, the pretrained Whisper model seems to basically perform well enough already on the LibriSpeech dataset, so any more training could potentially cause instability. I was only able to get a 0.5% increase in performance when looking at the normalized text.

**Comment by iamgroot42** [comment]
Thanks for the response!

Changing from capitalized to lower-case text leads to a much more stable learning curve (that somewhat matches your plots) and get the model to a WER slightly better than without fine-tuning (but only by a small percentage, as you mention). I should mention that I froze the encoder (as does the [code](https://github.com/Cabbagito/Fine-Tuning-Whisper-on-LibriSpeech/blob/1e794617e161f806021e922de00f5ea7cde11b96/training_script.py#L34) in this repository currently, I think?)
